proxy/src/main/java/com/velocitypowered/proxy/event/VelocityEventManager.java

```diff
@@ -103,12 +103,12 @@ public VelocityEventManager(final PluginManager pluginManager) {
   /**
    * Represents the registration of a single {@link EventHandler}.
    */
-  static final class HandlerRegistration {
+  static final class HandlerRegistration<E> {
 
     final PluginContainer plugin;
     final short order;
     final Class<?> eventType;
-    final EventHandler<Event> handler;
+    final EventHandler<E> handler;
```

Comment:
I prefer this changed back to `EventHandler<Object>` then and remove the generic, no need to pollute the impl with unnecessary generics.